python/hail/table.py

```diff
@@ -1768,8 +1781,25 @@ def rename(self, mapping):
         :class:`.Table`
             Table with renamed fields.
         """
-
-        return Table(self._jt.rename(mapping))
+        seen = {}
+
+        rowMap = {}
+        globalMap = {}
+
+        for k, v in mapping.items():
+            if v in seen:
+                raise ValueError(
+                    "Cannot rename two fields to the same name: attempted to rename '{}' and '{}' both to '{}'".format(
```

Comment:
instead of doing `'{}'` with str(k) and str(v), do `{}` with `repr(k)` and `repr(v)`. We should do this everywhere.